src/main/ngapp/accio/src/app/editor/editor.component.ts

```diff
@@ -10,54 +10,112 @@ import { imageUrls } from '../images';
-  image = new Image();
 
-  constructor(
-    private route: ActivatedRoute,
-    private router: Router,
-  ) { }
+  @ViewChild('hiddenCanvas', { static: true })
+  hiddenCanvas: ElementRef<HTMLCanvasElement>; 
+  private hiddenCtx: CanvasRenderingContext2D;
+
+  //  @Param scaleFactor is used to trim image scale so image 
+  //    is smaller than width and height of the users screen.
+  private scaleFactor = .9;
+  private image: HTMLImageElement;
+  private innerHeight: number;
+  private maskData: ImageData;
 
   ngOnInit() {
-    //returns url user clicked in gallery component
+    this.image = new Image();
+    this.innerHeight = window.innerHeight;
+    // Returns url user clicked in gallery component.
     this.route.paramMap.subscribe(params => {
       let index = Number(params.get('imgUrl'));
       this.checkImageUrl(index);
       this.url = imageUrls[index];
     });
 
     this.ctx = this.canvas.nativeElement.getContext('2d');
+    this.hiddenCtx = this.hiddenCanvas.nativeElement.getContext('2d');
     this.draw();
-  }  
+  }   
   
-  //  Checks that route for editor URL is in bounds of array of Urls and is a number
-  //  Redirects to gallery component if not
-  private checkImageUrl(index: number) {
+  /**
+   * Checks that route for editor URL is in bounds of array of Urls and is a number
+   * Redirects to gallery component if not, does not complete editor component
+   * index: the route from the editor url passed through gallery.ts 
+   */
+  private checkImageUrl(index: number): void {
     if (Number.isNaN(index) || index >= imageUrls.length || index < 0) {    
       console.log('Index: ' + index + 'Not a number or greater than number of images, returning to Gallery');
       this.router.navigate(['/gallery']);
     }
   }
-  
-  // Draws the image user selects from gallery on Canvas
-  // Currently, editor can require user to scroll to access
-  //  the entire photo, need to make it so the editor is 
-  //  fixed to screen size
-  private draw() {
+  /**  
+   * Draws the image user selects from gallery on Canvas
+   *    and creates a hidden canvas to store the original image 
+   *    as a reference when scaling the imageUI
+   * TODO(shmcaffrey): Currently, editor can require user to 
+   * scroll to access the entire photo, need to make it 
+   * so the editor is fixed to screen size.
+   */
+  private draw(): void {
     this.image.src = this.url;
 
-    //adjust canvas to image width and height
-    this.canvas.nativeElement.width = this.image.width;
-    this.canvas.nativeElement.height = this.image.height;
+    let imgWidth = this.image.width;
+    let imgHeight = this.image.height;
+
+    // Initialize hidden canvas width and height to original images width and height.
+    this.hiddenCanvas.nativeElement.width = imgWidth;
+    this.hiddenCanvas.nativeElement.height = imgHeight;
+
+    // Used to scale the image to the window size, 1.2 so the image is smaller than the window.
```

Comment:
nit: I don't know where the 1.2 is coming from in this comment.  It seems like might be a stale comment.  If not then maybe expand upon where the 1.2 is coming from.

Reply:
I also changed this comment?? I don't know why those changes never got pushed, I changed it to .9, the value of scaleFactor. This is just a factor to make the scaled image smaller than the users screen, .9/1.2 are both arbitrary, they could be any number.